### PR TITLE
Compiling, empty unit test suite

### PR DESCRIPTION
**Pull request by ghost**
- add a tasty unit test suite (should use auto-discover, but need to update travis first)
- allow for more tests to come

### CODE REVIEW

**Comment by kanishka-azimi** [comment]
Mulling over doctest. Regardless, I would like a tasty suite ready, for any tests that are too involved to express in doctest, so I see this PR as relevant for all future paths.

**Comment by kanishka-azimi** [comment]
Can you link to your preferred template for running doctest in other projects

File: hPDB.cabal [hunk 1]
```diff
@@ -90,3 +90,45 @@ Library
...
+    ghc-options: -msse2 
+  if flag(have-mmap)
+    build-depends: mmap
+    cpp-options: -DHAVE_MMAP
+  -- These optimization options change a lot for GHC >= 7.4.1
+  ghc-options: -fspec-constr-count=4 -O3 
+  if flag(have-text-format)
+    cpp-options: -DHAVE_TEXT_FORMAT
+    build-depends: text-format >= 0.3.1.0
+  other-extensions:       ScopedTypeVariables OverloadedStrings BangPatterns NoMonomorphismRestriction EmptyDataDecls MagicHash CPP PatternGuards NamedFieldPuns DisambiguateRecordFields TemplateHaskell MultiParamTypeClasses FlexibleInstances FlexibleContexts DeriveGeneric
```
Comment (ghost) [inline hPDB.cabal:130]: Flags relevant for test suite?

File: hPDB.cabal [hunk 2]
```diff
@@ -90,3 +90,45 @@ Library
...
+                    unordered-containers >= 0.2.5.0,
+                    deepseq,
+                    QuickCheck >= 2.5.0.0,
+                    text>=0.11.1.13,
+                    iterable >= 3.0,
+                    tagged >= 0.7,
+                    parallel >= 3.0.0.0,
+                    bytestring,
+                    zlib,
+                    Octree>= 0.6
```
Comment (ghost) [inline hPDB.cabal:119]: Remove most dependencies above until they are needed

**Comment by ghost** [comment]
Please reopen this PR when you are ready to review it or indicate an alternative approach that you would prefer (start with doctest? start w/validity?). Thanks!

**Pull request closed without merging**